Rudimentary navbar

alpuks opened the pull request:
Adds a rudimentary navbar, which takes authentication using the context already setup.

Fixed CSS conflict between tailwind and hero (beta version of hero required for tailwind v4)

Commit:
corrected tailwindcss and heroui for css to work

Commit:
added navbar with authentication based routing

Commit:
authentication via cookie (uses workaroud for cookie format)
hardcoded backend URL for testing
automatic redirect to SSO on accessing unauthenticated pages with redirect back to target page

alpuks commented:
backend returns cookie in a string format that is not JSON compliant (keys and values not in quotes), currently the workaround is manually splitting the string into key-value pairs and then parsing them

backend currently often fails IAT validation, which might be caused due to running in docker and having time drift with base OS (base IAT validation is with 1 sec accuracy) - for testing adding jwt.WithValidate(false) to jwk.go works

For authentication, it uses a hardcoded localhost:2727 link with passing the current location, so if you access a page where you need to be logged in, you'll be redirected directly to SSO and then back to the page

AlHeamer commented:
0697791 base64 encodes the json in the cookie. decoding maintains quotes. Apparently that's a cookie thing.

da5336a add 5 minute acceptable skew on jwt issued at field (also added `JWT_SKEW` env var for custom values)

Not sure I fully understand the 3rd point, is that related to the src value after login?

Commit:
base64 decoding authentication cookie

Commit:
login button uses SSO login image
removed hardcoding of backend URL

Commit:
formatting

Commit:
added theme toggle

Commit:
corrected heroUI elements to use tanstack navigation

Commit:
corrected eve sso image switch based on theme

alpuks commented:
Corrected cookie usage

Added light-dark theme toggle (using the built-in themes on all heroUI elements)

Switched login button into the eve SSO image

Commit:
changed navbar to match mock more
